recipes/szip/all/conanfile.py

```diff
@@ -28,91 +31,103 @@ class SzipConan(ConanFile):
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-        # TODO: to remove in conan v2 once cmake_find_package* generators removed
-        self._create_cmake_module_alias_targets(
-            os.path.join(self.package_folder, self._module_file_rel_path),
-            {"szip-shared" if self.options.shared else "szip-static": "szip::szip"}
-        )
-
-    @staticmethod
-    def _create_cmake_module_alias_targets(module_file, targets):
-        content = ""
-        for alias, aliased in targets.items():
-            content += textwrap.dedent("""\
-                if(TARGET {aliased} AND NOT TARGET {alias})
-                    add_library({alias} INTERFACE IMPORTED)
-                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
-                endif()
-            """.format(alias=alias, aliased=aliased))
-        tools.save(module_file, content)
-
-    @property
-    def _module_file_rel_path(self):
-        return os.path.join("lib", "cmake", "conan-official-{}-targets.cmake".format(self.name))
+# QUESTION FIXME is this required?
+#        # TODO: to remove in conan v2 once cmake_find_package* generators removed
+#        self._create_cmake_module_alias_targets(
+#            os.path.join(self.package_folder, self._module_file_rel_path),
+#            {"szip-shared" if self.options.shared else "szip-static": "szip::szip"}
+#        )
+#
+#    @staticmethod
+#    def _create_cmake_module_alias_targets(module_file, targets):
+#        content = ""
+#        for alias, aliased in targets.items():
+#            content += textwrap.dedent("""\
+#                if(TARGET {aliased} AND NOT TARGET {alias})
+#                    add_library({alias} INTERFACE IMPORTED)
+#                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+#                endif()
+#            """.format(alias=alias, aliased=aliased))
+#        tools.save(module_file, content)
+#
+#    @property
+#    def _module_file_rel_path(self):
+#        return os.path.join("lib", "cmake", "conan-official-{}-targets.cmake".format(self.name))
 
     def package_info(self):
+        libname = "szip"
+        if self.settings.build_type == "Debug":
+            libname = "szip_D" if self.settings.os == "Windows" else "szip_debug"
+        self.cpp_info.libs = [libname]
+
         self.cpp_info.set_property("cmake_file_name", "szip")
-        self.cpp_info.set_property("cmake_target_name", "szip-shared" if self.options.shared else "szip-static")
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.set_property("cmake_target_name", "szip::szip")
+
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs = ["m"]
+
+# QUESTION FIXME the old target was just "szip-shared" or "szip-static", should we keep supporting this, and how?
```

Comment:
Please, remove those comments. Conan V1 is still mandatory.

Reply:
ok